service/src/main/java/gov/nasa/pds/api/engineering/elasticsearch/business/RequestAndResponseContext.java

```diff
@@ -161,10 +161,17 @@ private RequestAndResponseContext(
 		}
 		else
 		{
+			String known = "";
 			log.warn("Could not find a matach for application type: " + String.valueOf(this.format));
 			log.warn("   Known types: " + String.valueOf(this.formatters.keySet().size()));
-			for (String key : this.formatters.keySet()) log.warn("      key: " + String.valueOf(key));
-			throw new ApplicationTypeException("The given application type, " + String.valueOf(this.format) + ", is not known by RquestAndResponseContext.");
+			for (String key : this.formatters.keySet())
+			{
+				log.warn("      key: " + String.valueOf(key));
```

Comment:
Do we need a warning for this message ? I would think of DEBUG, unless I misunderstood what it does.

Reply:
Need in what way? Will the program work without the warning, yes. When a user says they got a 406 on Saturday will you be able to find it in the log if not posted, no. So define what you mean by need.

Warnings are usually issued when the program detects something fishy (a handled exception) notifying the initiator of the fishy bit that something looks wrong but was handled albeit maybe not how the initiator wanted so they may want to try again -- think or google compiler warnings. Warnings differs from errors in that the program does not crash and burn. Debug on the other hand is way of tracing the program to determine where a bug is hiding or may exist or prove its existance.

For this message when a user mentions days later that a 406 was returned (something fishy but handled) we would have the appropriate level of logging.

If you want it to be debug then simply say so directly and I will change it. One could argue that given errors are being returned to the user the logging is superfluous.